### PR TITLE
Use pysha3 to compute function signature from string

### PR DESCRIPTION
**Pull request by ClementWalter**
## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Tests include massive magic strings.

Issue Number: N/A

## What is the new behavior?

Using pysha3 to compute the function selector from the function name hopefully gives more context to the calldata.

## Other information

We might to use a more advanced lib targeting evm dev in python. I didn't look at this as I just wanted to better understand what was actually the parameters of the tests.

### CODE REVIEW

**Review by abdelhamidbakhta — APPROVED** [review]
lgtm

File: tests/integrations/test_cases/execute_at_address.py [hunk 1]
```diff
@@ -29,24 +41,12 @@
...
-            "calldata": "18160ddd",
+            "calldata": keccak_256("totalSupply()".encode()).hexdigest()[:8],
             "stack": "",
             "memory": "",
             "return_value": "0000000000000000000000000000000000000000000000000000000000000000",
         },
         "id": "solmate_erc20_totalSupply",
         "marks": [pytest.mark.SolmateERC20],
     },
-    {
```
Comment (Eikix) [inline tests/integrations/test_cases/execute_at_address.py:40]: Why did you delete this?

Comment (ClementWalter) [reply tests/integrations/test_cases/execute_at_address.py:40]: furthermore, in the context of `execute_at_address`, it should revert as the calldata is not long enough (2 bytes instead of 4)

Comment (Eikix) [reply tests/integrations/test_cases/execute_at_address.py:40]: right!

**Review by Eikix — APPROVED** [review]
